// @uploadfly/react-widget/dist/react-widget.js
import H, { useState as G, useRef as Tr } from "react";
function Cr(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Be = { exports: {} }, te = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var yt;
function Pr() {
  if (yt)
    return te;
  yt = 1;
  var e = H, t = Symbol.for("react.element"), r = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, o = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function a(c, p, u) {
    var d, y = {}, w = null, R = null;
    u !== void 0 && (w = "" + u), p.key !== void 0 && (w = "" + p.key), p.ref !== void 0 && (R = p.ref);
    for (d in p)
      n.call(p, d) && !i.hasOwnProperty(d) && (y[d] = p[d]);
    if (c && c.defaultProps)
      for (d in p = c.defaultProps, p)
        y[d] === void 0 && (y[d] = p[d]);
    return { $$typeof: t, type: c, key: w, ref: R, props: y, _owner: o.current };
  }
  return te.Fragment = r, te.jsx = a, te.jsxs = a, te;
}
var re = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Et;
function Ar() {
  return Et || (Et = 1, process.env.NODE_ENV !== "production" && function() {
    var e = H, t = Symbol.for("react.element"), r = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), a = Symbol.for("react.provider"), c = Symbol.for("react.context"), p = Symbol.for("react.forward_ref"), u = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), y = Symbol.for("react.memo"), w = Symbol.for("react.lazy"), R = Symbol.for("react.offscreen"), m = Symbol.iterator, b = "@@iterator";
    function D(s) {
      if (s === null || typeof s != "object")
        return null;
      var f = m && s[m] || s[b];
      return typeof f == "function" ? f : null;
    }
    var C = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function T(s) {
      {
        for (var f = arguments.length, h = new Array(f > 1 ? f - 1 : 0), E = 1; E < f; E++)
          h[E - 1] = arguments[E];
        L("error", s, h);
      }
    }
    function L(s, f, h) {
      {
        var E = C.ReactDebugCurrentFrame, S = E.getStackAddendum();
        S !== "" && (f += "%s", h = h.concat([S]));
        var x = h.map(function(v) {
          return String(v);
        });
        x.unshift("Warning: " + f), Function.prototype.apply.call(console[s], console, x);
      }
    }
    var N = !1, J = !1, Xt = !1, Zt = !1, Qt = !1, Ve;
    Ve = Symbol.for("react.module.reference");
    function er(s) {
      return !!(typeof s == "string" || typeof s == "function" || s === n || s === i || Qt || s === o || s === u || s === d || Zt || s === R || N || J || Xt || typeof s == "object" && s !== null && (s.$$typeof === w || s.$$typeof === y || s.$$typeof === a || s.$$typeof === c || s.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      s.$$typeof === Ve || s.getModuleId !== void 0));
    }
    function tr(s, f, h) {
      var E = s.displayName;
      if (E)
        return E;
      var S = f.displayName || f.name || "";
      return S !== "" ? h + "(" + S + ")" : h;
    }
    function Je(s) {
      return s.displayName || "Context";
    }
    function M(s) {
      if (s == null)
        return null;
      if (typeof s.tag == "number" && T("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof s == "function")
        return s.displayName || s.name || null;
      if (typeof s == "string")
        return s;
      switch (s) {
        case n:
          return "Fragment";
        case r:
          return "Portal";
        case i:
          return "Profiler";
        case o:
          return "StrictMode";
        case u:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case c:
            var f = s;
            return Je(f) + ".Consumer";
          case a:
            var h = s;
            return Je(h._context) + ".Provider";
          case p:
            return tr(s, s.render, "ForwardRef");
          case y:
            var E = s.displayName || null;
            return E !== null ? E : M(s.type) || "Memo";
          case w: {
            var S = s, x = S._payload, v = S._init;
            try {
              return M(v(x));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var V = Object.assign, Q = 0, Ye, Ke, Ge, Xe, Ze, Qe, et;
    function tt() {
    }
    tt.__reactDisabledLog = !0;
    function rr() {
      {
        if (Q === 0) {
          Ye = console.log, Ke = console.info, Ge = console.warn, Xe = console.error, Ze = console.group, Qe = console.groupCollapsed, et = console.groupEnd;
          var s = {
            configurable: !0,
            enumerable: !0,
            value: tt,
            writable: !0
          };
          Object.defineProperties(console, {
            info: s,
            log: s,
            warn: s,
            error: s,
            group: s,
            groupCollapsed: s,
            groupEnd: s
          });
        }
        Q++;
      }
    }
    function nr() {
      {
        if (Q--, Q === 0) {
          var s = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: V({}, s, {
              value: Ye
            }),
            info: V({}, s, {
              value: Ke
            }),
            warn: V({}, s, {
              value: Ge
            }),
            error: V({}, s, {
              value: Xe
            }),
            group: V({}, s, {
              value: Ze
            }),
            groupCollapsed: V({}, s, {
              value: Qe
            }),
            groupEnd: V({}, s, {
              value: et
            })
          });
        }
        Q < 0 && T("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var Oe = C.ReactCurrentDispatcher, Se;
    function le(s, f, h) {
      {
        if (Se === void 0)
          try {
            throw Error();
          } catch (S) {
            var E = S.stack.trim().match(/\n( *(at )?)/);
            Se = E && E[1] || "";
          }
        return `
` + Se + s;
      }
    }
    var xe = !1, ue;
    {
      var sr = typeof WeakMap == "function" ? WeakMap : Map;
      ue = new sr();
    }
    function rt(s, f) {
      if (!s || xe)
        return "";
      {
        var h = ue.get(s);
        if (h !== void 0)
          return h;
      }
      var E;
      xe = !0;
      var S = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var x;
      x = Oe.current, Oe.current = null, rr();
      try {
        if (f) {
          var v = function() {
            throw Error();
          };
          if (Object.defineProperty(v.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(v, []);
            } catch (z) {
              E = z;
            }
            Reflect.construct(s, [], v);
          } else {
            try {
              v.call();
            } catch (z) {
              E = z;
            }
            s.call(v.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (z) {
            E = z;
          }
          s();
        }
      } catch (z) {
        if (z && E && typeof z.stack == "string") {
          for (var g = z.stack.split(`
`), F = E.stack.split(`
`), P = g.length - 1, A = F.length - 1; P >= 1 && A >= 0 && g[P] !== F[A]; )
            A--;
          for (; P >= 1 && A >= 0; P--, A--)
            if (g[P] !== F[A]) {
              if (P !== 1 || A !== 1)
                do
                  if (P--, A--, A < 0 || g[P] !== F[A]) {
                    var k = `
` + g[P].replace(" at new ", " at ");
                    return s.displayName && k.includes("<anonymous>") && (k = k.replace("<anonymous>", s.displayName)), typeof s == "function" && ue.set(s, k), k;
                  }
                while (P >= 1 && A >= 0);
              break;
            }
        }
      } finally {
        xe = !1, Oe.current = x, nr(), Error.prepareStackTrace = S;
      }
      var K = s ? s.displayName || s.name : "", mt = K ? le(K) : "";
      return typeof s == "function" && ue.set(s, mt), mt;
    }
    function or(s, f, h) {
      return rt(s, !1);
    }
    function ir(s) {
      var f = s.prototype;
      return !!(f && f.isReactComponent);
    }
    function ce(s, f, h) {
      if (s == null)
        return "";
      if (typeof s == "function")
        return rt(s, ir(s));
      if (typeof s == "string")
        return le(s);
      switch (s) {
        case u:
          return le("Suspense");
        case d:
          return le("SuspenseList");
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case p:
            return or(s.render);
          case y:
            return ce(s.type, f, h);
          case w: {
            var E = s, S = E._payload, x = E._init;
            try {
              return ce(x(S), f, h);
            } catch {
            }
          }
        }
      return "";
    }
    var fe = Object.prototype.hasOwnProperty, nt = {}, st = C.ReactDebugCurrentFrame;
    function de(s) {
      if (s) {
        var f = s._owner, h = ce(s.type, s._source, f ? f.type : null);
        st.setExtraStackFrame(h);
      } else
        st.setExtraStackFrame(null);
    }
    function ar(s, f, h, E, S) {
      {
        var x = Function.call.bind(fe);
        for (var v in s)
          if (x(s, v)) {
            var g = void 0;
            try {
              if (typeof s[v] != "function") {
                var F = Error((E || "React class") + ": " + h + " type `" + v + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof s[v] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw F.name = "Invariant Violation", F;
              }
              g = s[v](f, v, E, h, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (P) {
              g = P;
            }
            g && !(g instanceof Error) && (de(S), T("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", E || "React class", h, v, typeof g), de(null)), g instanceof Error && !(g.message in nt) && (nt[g.message] = !0, de(S), T("Failed %s type: %s", h, g.message), de(null));
          }
      }
    }
    var lr = Array.isArray;
    function Te(s) {
      return lr(s);
    }
    function ur(s) {
      {
        var f = typeof Symbol == "function" && Symbol.toStringTag, h = f && s[Symbol.toStringTag] || s.constructor.name || "Object";
        return h;
      }
    }
    function cr(s) {
      try {
        return ot(s), !1;
      } catch {
        return !0;
      }
    }
    function ot(s) {
      return "" + s;
    }
    function it(s) {
      if (cr(s))
        return T("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ur(s)), ot(s);
    }
    var ee = C.ReactCurrentOwner, fr = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, at, lt, Ce;
    Ce = {};
    function dr(s) {
      if (fe.call(s, "ref")) {
        var f = Object.getOwnPropertyDescriptor(s, "ref").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return s.ref !== void 0;
    }
    function pr(s) {
      if (fe.call(s, "key")) {
        var f = Object.getOwnPropertyDescriptor(s, "key").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return s.key !== void 0;
    }
    function hr(s, f) {
      if (typeof s.ref == "string" && ee.current && f && ee.current.stateNode !== f) {
        var h = M(ee.current.type);
        Ce[h] || (T('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', M(ee.current.type), s.ref), Ce[h] = !0);
      }
    }
    function mr(s, f) {
      {
        var h = function() {
          at || (at = !0, T("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        h.isReactWarning = !0, Object.defineProperty(s, "key", {
          get: h,
          configurable: !0
        });
      }
    }
    function yr(s, f) {
      {
        var h = function() {
          lt || (lt = !0, T("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        h.isReactWarning = !0, Object.defineProperty(s, "ref", {
          get: h,
          configurable: !0
        });
      }
    }
    var Er = function(s, f, h, E, S, x, v) {
      var g = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: s,
        key: f,
        ref: h,
        props: v,
        // Record the component responsible for creating this element.
        _owner: x
      };
      return g._store = {}, Object.defineProperty(g._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(g, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: E
      }), Object.defineProperty(g, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: S
      }), Object.freeze && (Object.freeze(g.props), Object.freeze(g)), g;
    };
    function br(s, f, h, E, S) {
      {
        var x, v = {}, g = null, F = null;
        h !== void 0 && (it(h), g = "" + h), pr(f) && (it(f.key), g = "" + f.key), dr(f) && (F = f.ref, hr(f, S));
        for (x in f)
          fe.call(f, x) && !fr.hasOwnProperty(x) && (v[x] = f[x]);
        if (s && s.defaultProps) {
          var P = s.defaultProps;
          for (x in P)
            v[x] === void 0 && (v[x] = P[x]);
        }
        if (g || F) {
          var A = typeof s == "function" ? s.displayName || s.name || "Unknown" : s;
          g && mr(v, A), F && yr(v, A);
        }
        return Er(s, g, F, S, E, ee.current, v);
      }
    }
    var Pe = C.ReactCurrentOwner, ut = C.ReactDebugCurrentFrame;
    function Y(s) {
      if (s) {
        var f = s._owner, h = ce(s.type, s._source, f ? f.type : null);
        ut.setExtraStackFrame(h);
      } else
        ut.setExtraStackFrame(null);
    }
    var Ae;
    Ae = !1;
    function je(s) {
      return typeof s == "object" && s !== null && s.$$typeof === t;
    }
    function ct() {
      {
        if (Pe.current) {
          var s = M(Pe.current.type);
          if (s)
            return `

Check the render method of \`` + s + "`.";
        }
        return "";
      }
    }
    function gr(s) {
      {
        if (s !== void 0) {
          var f = s.fileName.replace(/^.*[\\\/]/, ""), h = s.lineNumber;
          return `

Check your code at ` + f + ":" + h + ".";
        }
        return "";
      }
    }
    var ft = {};
    function vr(s) {
      {
        var f = ct();
        if (!f) {
          var h = typeof s == "string" ? s : s.displayName || s.name;
          h && (f = `

Check the top-level render call using <` + h + ">.");
        }
        return f;
      }
    }
    function dt(s, f) {
      {
        if (!s._store || s._store.validated || s.key != null)
          return;
        s._store.validated = !0;
        var h = vr(f);
        if (ft[h])
          return;
        ft[h] = !0;
        var E = "";
        s && s._owner && s._owner !== Pe.current && (E = " It was passed a child from " + M(s._owner.type) + "."), Y(s), T('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', h, E), Y(null);
      }
    }
    function pt(s, f) {
      {
        if (typeof s != "object")
          return;
        if (Te(s))
          for (var h = 0; h < s.length; h++) {
            var E = s[h];
            je(E) && dt(E, f);
          }
        else if (je(s))
          s._store && (s._store.validated = !0);
        else if (s) {
          var S = D(s);
          if (typeof S == "function" && S !== s.entries)
            for (var x = S.call(s), v; !(v = x.next()).done; )
              je(v.value) && dt(v.value, f);
        }
      }
    }
    function wr(s) {
      {
        var f = s.type;
        if (f == null || typeof f == "string")
          return;
        var h;
        if (typeof f == "function")
          h = f.propTypes;
        else if (typeof f == "object" && (f.$$typeof === p || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        f.$$typeof === y))
          h = f.propTypes;
        else
          return;
        if (h) {
          var E = M(f);
          ar(h, s.props, "prop", E, s);
        } else if (f.PropTypes !== void 0 && !Ae) {
          Ae = !0;
          var S = M(f);
          T("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", S || "Unknown");
        }
        typeof f.getDefaultProps == "function" && !f.getDefaultProps.isReactClassApproved && T("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Rr(s) {
      {
        for (var f = Object.keys(s.props), h = 0; h < f.length; h++) {
          var E = f[h];
          if (E !== "children" && E !== "key") {
            Y(s), T("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", E), Y(null);
            break;
          }
        }
        s.ref !== null && (Y(s), T("Invalid attribute `ref` supplied to `React.Fragment`."), Y(null));
      }
    }
    function ht(s, f, h, E, S, x) {
      {
        var v = er(s);
        if (!v) {
          var g = "";
          (s === void 0 || typeof s == "object" && s !== null && Object.keys(s).length === 0) && (g += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var F = gr(S);
          F ? g += F : g += ct();
          var P;
          s === null ? P = "null" : Te(s) ? P = "array" : s !== void 0 && s.$$typeof === t ? (P = "<" + (M(s.type) || "Unknown") + " />", g = " Did you accidentally export a JSX literal instead of a component?") : P = typeof s, T("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", P, g);
        }
        var A = br(s, f, h, S, x);
        if (A == null)
          return A;
        if (v) {
          var k = f.children;
          if (k !== void 0)
            if (E)
              if (Te(k)) {
                for (var K = 0; K < k.length; K++)
                  pt(k[K], s);
                Object.freeze && Object.freeze(k);
              } else
                T("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              pt(k, s);
        }
        return s === n ? Rr(A) : wr(A), A;
      }
    }
    function _r(s, f, h) {
      return ht(s, f, h, !0);
    }
    function Or(s, f, h) {
      return ht(s, f, h, !1);
    }
    var Sr = Or, xr = _r;
    re.Fragment = n, re.jsx = Sr, re.jsxs = xr;
  }()), re;
}
process.env.NODE_ENV === "production" ? Be.exports = Pr() : Be.exports = Ar();
var O = Be.exports, Pt = {
  color: void 0,
  size: void 0,
  className: void 0,
  style: void 0,
  attr: void 0
}, bt = H.createContext && H.createContext(Pt), W = globalThis && globalThis.__assign || function() {
  return W = Object.assign || function(e) {
    for (var t, r = 1, n = arguments.length; r < n; r++) {
      t = arguments[r];
      for (var o in t)
        Object.prototype.hasOwnProperty.call(t, o) && (e[o] = t[o]);
    }
    return e;
  }, W.apply(this, arguments);
}, jr = globalThis && globalThis.__rest || function(e, t) {
  var r = {};
  for (var n in e)
    Object.prototype.hasOwnProperty.call(e, n) && t.indexOf(n) < 0 && (r[n] = e[n]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var o = 0, n = Object.getOwnPropertySymbols(e); o < n.length; o++)
      t.indexOf(n[o]) < 0 && Object.prototype.propertyIsEnumerable.call(e, n[o]) && (r[n[o]] = e[n[o]]);
  return r;
};
function At(e) {
  return e && e.map(function(t, r) {
    return H.createElement(t.tag, W({
      key: r
    }, t.attr), At(t.child));
  });
}
function oe(e) {
  return function(t) {
    return H.createElement(Nr, W({
      attr: W({}, e.attr)
    }, t), At(e.child));
  };
}
function Nr(e) {
  var t = function(r) {
    var n = e.attr, o = e.size, i = e.title, a = jr(e, ["attr", "size", "title"]), c = o || r.size || "1em", p;
    return r.className && (p = r.className), e.className && (p = (p ? p + " " : "") + e.className), H.createElement("svg", W({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, r.attr, n, a, {
      className: p,
      style: W(W({
        color: e.color || r.color
      }, r.style), e.style),
      height: c,
      width: c,
      xmlns: "http://www.w3.org/2000/svg"
    }), i && H.createElement("title", null, i), e.children);
  };
  return bt !== void 0 ? H.createElement(bt.Consumer, null, function(r) {
    return t(r);
  }) : t(Pt);
}
function Fr(e) {
  return oe({ tag: "svg", attr: { viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { fill: "none", d: "M0 0h24v24H0z" } }, { tag: "path", attr: { d: "M5 20h14v-2H5v2zm0-10h4v6h6v-6h4l-7-7-7 7z" } }] })(e);
}
function Dr(e) {
  return oe({ tag: "svg", attr: { viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { fill: "none", d: "M0 0h24v24H0z" } }, { tag: "path", attr: { d: "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" } }] })(e);
}
var jt = { exports: {} };
(function(e) {
  (function(t) {
    if (e.exports)
      return e.exports = t();
    this.filesize = t();
  })(function() {
    var t = "BKMGTPEZY".split("");
    function r(n, o) {
      return n && n.toLowerCase() === o.toLowerCase();
    }
    return function(o, i) {
      return o = typeof o == "number" ? o : 0, i = i || {}, i.fixed = typeof i.fixed == "number" ? i.fixed : 2, i.spacer = typeof i.spacer == "string" ? i.spacer : " ", i.calculate = function(a) {
        var c = r(a, "si") ? ["k", "B"] : ["K", "iB"], p = r(a, "si") ? 1e3 : 1024, u = Math.log(o) / Math.log(p) | 0, d = o / Math.pow(p, u), y = d.toFixed(i.fixed), w;
        return u - 1 < 3 && !r(a, "si") && r(a, "jedec") && (c[1] = "B"), w = u ? (c[0] + "MGTPEZY")[u - 1] + c[1] : (y | 0) === 1 ? "Byte" : "Bytes", {
          suffix: w,
          magnitude: u,
          result: d,
          fixed: y,
          bits: { result: d / 8, fixed: (d / 8).toFixed(i.fixed) }
        };
      }, i.to = function(a, c) {
        var p = r(c, "si") ? 1e3 : 1024, u = t.indexOf(typeof a == "string" ? a[0].toUpperCase() : "B"), d = o;
        if (u === -1 || u === 0)
          return d.toFixed(2);
        for (; u > 0; u--)
          d /= p;
        return d.toFixed(2);
      }, i.human = function(a) {
        var c = i.calculate(a);
        return c.fixed + i.spacer + c.suffix;
      }, i;
    };
  });
})(jt);
var kr = jt.exports;
const Br = /* @__PURE__ */ Cr(kr);
function Nt(e, t) {
  return function() {
    return e.apply(t, arguments);
  };
}
const { toString: Lr } = Object.prototype, { getPrototypeOf: ze } = Object, be = ((e) => (t) => {
  const r = Lr.call(t);
  return e[r] || (e[r] = r.slice(8, -1).toLowerCase());
})(/* @__PURE__ */ Object.create(null)), I = (e) => (e = e.toLowerCase(), (t) => be(t) === e), ge = (e) => (t) => typeof t === e, { isArray: Z } = Array, se = ge("undefined");
function Ur(e) {
  return e !== null && !se(e) && e.constructor !== null && !se(e.constructor) && B(e.constructor.isBuffer) && e.constructor.isBuffer(e);
}
const Ft = I("ArrayBuffer");
function Ir(e) {
  let t;
  return typeof ArrayBuffer < "u" && ArrayBuffer.isView ? t = ArrayBuffer.isView(e) : t = e && e.buffer && Ft(e.buffer), t;
}
const Mr = ge("string"), B = ge("function"), Dt = ge("number"), ve = (e) => e !== null && typeof e == "object", zr = (e) => e === !0 || e === !1, pe = (e) => {
  if (be(e) !== "object")
    return !1;
  const t = ze(e);
  return (t === null || t === Object.prototype || Object.getPrototypeOf(t) === null) && !(Symbol.toStringTag in e) && !(Symbol.iterator in e);
}, Hr = I("Date"), $r = I("File"), qr = I("Blob"), Wr = I("FileList"), Vr = (e) => ve(e) && B(e.pipe), Jr = (e) => {
  let t;
  return e && (typeof FormData == "function" && e instanceof FormData || B(e.append) && ((t = be(e)) === "formdata" || // detect form-data instance
  t === "object" && B(e.toString) && e.toString() === "[object FormData]"));
}, Yr = I("URLSearchParams"), Kr = (e) => e.trim ? e.trim() : e.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, "");
function ie(e, t, { allOwnKeys: r = !1 } = {}) {
  if (e === null || typeof e > "u")
    return;
  let n, o;
  if (typeof e != "object" && (e = [e]), Z(e))
    for (n = 0, o = e.length; n < o; n++)
      t.call(null, e[n], n, e);
  else {
    const i = r ? Object.getOwnPropertyNames(e) : Object.keys(e), a = i.length;
    let c;
    for (n = 0; n < a; n++)
      c = i[n], t.call(null, e[c], c, e);
  }
}
function kt(e, t) {
  t = t.toLowerCase();
  const r = Object.keys(e);
  let n = r.length, o;
  for (; n-- > 0; )
    if (o = r[n], t === o.toLowerCase())
      return o;
  return null;
}
const Bt = (() => typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : global)(), Lt = (e) => !se(e) && e !== Bt;
function Le() {
  const { caseless: e } = Lt(this) && this || {}, t = {}, r = (n, o) => {
    const i = e && kt(t, o) || o;
    pe(t[i]) && pe(n) ? t[i] = Le(t[i], n) : pe(n) ? t[i] = Le({}, n) : Z(n) ? t[i] = n.slice() : t[i] = n;
  };
  for (let n = 0, o = arguments.length; n < o; n++)
    arguments[n] && ie(arguments[n], r);
  return t;
}
const Gr = (e, t, r, { allOwnKeys: n } = {}) => (ie(t, (o, i) => {
  r && B(o) ? e[i] = Nt(o, r) : e[i] = o;
}, { allOwnKeys: n }), e), Xr = (e) => (e.charCodeAt(0) === 65279 && (e = e.slice(1)), e), Zr = (e, t, r, n) => {
  e.prototype = Object.create(t.prototype, n), e.prototype.constructor = e, Object.defineProperty(e, "super", {
    value: t.prototype
  }), r && Object.assign(e.prototype, r);
}, Qr = (e, t, r, n) => {
  let o, i, a;
  const c = {};
  if (t = t || {}, e == null)
    return t;
  do {
    for (o = Object.getOwnPropertyNames(e), i = o.length; i-- > 0; )
      a = o[i], (!n || n(a, e, t)) && !c[a] && (t[a] = e[a], c[a] = !0);
    e = r !== !1 && ze(e);
  } while (e && (!r || r(e, t)) && e !== Object.prototype);
  return t;
}, en = (e, t, r) => {
  e = String(e), (r === void 0 || r > e.length) && (r = e.length), r -= t.length;
  const n = e.indexOf(t, r);
  return n !== -1 && n === r;
}, tn = (e) => {
  if (!e)
    return null;
  if (Z(e))
    return e;
  let t = e.length;
  if (!Dt(t))
    return null;
  const r = new Array(t);
  for (; t-- > 0; )
    r[t] = e[t];
  return r;
}, rn = ((e) => (t) => e && t instanceof e)(typeof Uint8Array < "u" && ze(Uint8Array)), nn = (e, t) => {
  const n = (e && e[Symbol.iterator]).call(e);
  let o;
  for (; (o = n.next()) && !o.done; ) {
    const i = o.value;
    t.call(e, i[0], i[1]);
  }
}, sn = (e, t) => {
  let r;
  const n = [];
  for (; (r = e.exec(t)) !== null; )
    n.push(r);
  return n;
}, on = I("HTMLFormElement"), an = (e) => e.toLowerCase().replace(
  /[-_\s]([a-z\d])(\w*)/g,
  function(r, n, o) {
    return n.toUpperCase() + o;
  }
), gt = (({ hasOwnProperty: e }) => (t, r) => e.call(t, r))(Object.prototype), ln = I("RegExp"), Ut = (e, t) => {
  const r = Object.getOwnPropertyDescriptors(e), n = {};
  ie(r, (o, i) => {
    t(o, i, e) !== !1 && (n[i] = o);
  }), Object.defineProperties(e, n);
}, un = (e) => {
  Ut(e, (t, r) => {
    if (B(e) && ["arguments", "caller", "callee"].indexOf(r) !== -1)
      return !1;
    const n = e[r];
    if (B(n)) {
      if (t.enumerable = !1, "writable" in t) {
        t.writable = !1;
        return;
      }
      t.set || (t.set = () => {
        throw Error("Can not rewrite read-only method '" + r + "'");
      });
    }
  });
}, cn = (e, t) => {
  const r = {}, n = (o) => {
    o.forEach((i) => {
      r[i] = !0;
    });
  };
  return Z(e) ? n(e) : n(String(e).split(t)), r;
}, fn = () => {
}, dn = (e, t) => (e = +e, Number.isFinite(e) ? e : t), Ne = "abcdefghijklmnopqrstuvwxyz", vt = "0123456789", It = {
  DIGIT: vt,
  ALPHA: Ne,
  ALPHA_DIGIT: Ne + Ne.toUpperCase() + vt
}, pn = (e = 16, t = It.ALPHA_DIGIT) => {
  let r = "";
  const { length: n } = t;
  for (; e--; )
    r += t[Math.random() * n | 0];
  return r;
};
function hn(e) {
  return !!(e && B(e.append) && e[Symbol.toStringTag] === "FormData" && e[Symbol.iterator]);
}
const mn = (e) => {
  const t = new Array(10), r = (n, o) => {
    if (ve(n)) {
      if (t.indexOf(n) >= 0)
        return;
      if (!("toJSON" in n)) {
        t[o] = n;
        const i = Z(n) ? [] : {};
        return ie(n, (a, c) => {
          const p = r(a, o + 1);
          !se(p) && (i[c] = p);
        }), t[o] = void 0, i;
      }
    }
    return n;
  };
  return r(e, 0);
}, yn = I("AsyncFunction"), En = (e) => e && (ve(e) || B(e)) && B(e.then) && B(e.catch), l = {
  isArray: Z,
  isArrayBuffer: Ft,
  isBuffer: Ur,
  isFormData: Jr,
  isArrayBufferView: Ir,
  isString: Mr,
  isNumber: Dt,
  isBoolean: zr,
  isObject: ve,
  isPlainObject: pe,
  isUndefined: se,
  isDate: Hr,
  isFile: $r,
  isBlob: qr,
  isRegExp: ln,
  isFunction: B,
  isStream: Vr,
  isURLSearchParams: Yr,
  isTypedArray: rn,
  isFileList: Wr,
  forEach: ie,
  merge: Le,
  extend: Gr,
  trim: Kr,
  stripBOM: Xr,
  inherits: Zr,
  toFlatObject: Qr,
  kindOf: be,
  kindOfTest: I,
  endsWith: en,
  toArray: tn,
  forEachEntry: nn,
  matchAll: sn,
  isHTMLForm: on,
  hasOwnProperty: gt,
  hasOwnProp: gt,
  // an alias to avoid ESLint no-prototype-builtins detection
  reduceDescriptors: Ut,
  freezeMethods: un,
  toObjectSet: cn,
  toCamelCase: an,
  noop: fn,
  toFiniteNumber: dn,
  findKey: kt,
  global: Bt,
  isContextDefined: Lt,
  ALPHABET: It,
  generateString: pn,
  isSpecCompliantForm: hn,
  toJSONObject: mn,
  isAsyncFn: yn,
  isThenable: En
};
function _(e, t, r, n, o) {
  Error.call(this), Error.captureStackTrace ? Error.captureStackTrace(this, this.constructor) : this.stack = new Error().stack, this.message = e, this.name = "AxiosError", t && (this.code = t), r && (this.config = r), n && (this.request = n), o && (this.response = o);
}
l.inherits(_, Error, {
  toJSON: function() {
    return {
      // Standard
      message: this.message,
      name: this.name,
      // Microsoft
      description: this.description,
      number: this.number,
      // Mozilla
      fileName: this.fileName,
      lineNumber: this.lineNumber,
      columnNumber: this.columnNumber,
      stack: this.stack,
      // Axios
      config: l.toJSONObject(this.config),
      code: this.code,
      status: this.response && this.response.status ? this.response.status : null
    };
  }
});
const Mt = _.prototype, zt = {};
[
  "ERR_BAD_OPTION_VALUE",
  "ERR_BAD_OPTION",
  "ECONNABORTED",
  "ETIMEDOUT",
  "ERR_NETWORK",
  "ERR_FR_TOO_MANY_REDIRECTS",
  "ERR_DEPRECATED",
  "ERR_BAD_RESPONSE",
  "ERR_BAD_REQUEST",
  "ERR_CANCELED",
  "ERR_NOT_SUPPORT",
  "ERR_INVALID_URL"
  // eslint-disable-next-line func-names
].forEach((e) => {
  zt[e] = { value: e };
});
Object.defineProperties(_, zt);
Object.defineProperty(Mt, "isAxiosError", { value: !0 });
_.from = (e, t, r, n, o, i) => {
  const a = Object.create(Mt);
  return l.toFlatObject(e, a, function(p) {
    return p !== Error.prototype;
  }, (c) => c !== "isAxiosError"), _.call(a, e.message, t, r, n, o), a.cause = e, a.name = e.name, i && Object.assign(a, i), a;
};
const bn = null;
function Ue(e) {
  return l.isPlainObject(e) || l.isArray(e);
}
function Ht(e) {
  return l.endsWith(e, "[]") ? e.slice(0, -2) : e;
}
function wt(e, t, r) {
  return e ? e.concat(t).map(function(o, i) {
    return o = Ht(o), !r && i ? "[" + o + "]" : o;
  }).join(r ? "." : "") : t;
}
function gn(e) {
  return l.isArray(e) && !e.some(Ue);
}
const vn = l.toFlatObject(l, {}, null, function(t) {
  return /^is[A-Z]/.test(t);
});
function we(e, t, r) {
  if (!l.isObject(e))
    throw new TypeError("target must be an object");
  t = t || new FormData(), r = l.toFlatObject(r, {
    metaTokens: !0,
    dots: !1,
    indexes: !1
  }, !1, function(b, D) {
    return !l.isUndefined(D[b]);
  });
  const n = r.metaTokens, o = r.visitor || d, i = r.dots, a = r.indexes, p = (r.Blob || typeof Blob < "u" && Blob) && l.isSpecCompliantForm(t);
  if (!l.isFunction(o))
    throw new TypeError("visitor must be a function");
  function u(m) {
    if (m === null)
      return "";
    if (l.isDate(m))
      return m.toISOString();
    if (!p && l.isBlob(m))
      throw new _("Blob is not supported. Use a Buffer instead.");
    return l.isArrayBuffer(m) || l.isTypedArray(m) ? p && typeof Blob == "function" ? new Blob([m]) : Buffer.from(m) : m;
  }
  function d(m, b, D) {
    let C = m;
    if (m && !D && typeof m == "object") {
      if (l.endsWith(b, "{}"))
        b = n ? b : b.slice(0, -2), m = JSON.stringify(m);
      else if (l.isArray(m) && gn(m) || (l.isFileList(m) || l.endsWith(b, "[]")) && (C = l.toArray(m)))
        return b = Ht(b), C.forEach(function(L, N) {
          !(l.isUndefined(L) || L === null) && t.append(
            // eslint-disable-next-line no-nested-ternary
            a === !0 ? wt([b], N, i) : a === null ? b : b + "[]",
            u(L)
          );
        }), !1;
    }
    return Ue(m) ? !0 : (t.append(wt(D, b, i), u(m)), !1);
  }
  const y = [], w = Object.assign(vn, {
    defaultVisitor: d,
    convertValue: u,
    isVisitable: Ue
  });
  function R(m, b) {
    if (!l.isUndefined(m)) {
      if (y.indexOf(m) !== -1)
        throw Error("Circular reference detected in " + b.join("."));
      y.push(m), l.forEach(m, function(C, T) {
        (!(l.isUndefined(C) || C === null) && o.call(
          t,
          C,
          l.isString(T) ? T.trim() : T,
          b,
          w
        )) === !0 && R(C, b ? b.concat(T) : [T]);
      }), y.pop();
    }
  }
  if (!l.isObject(e))
    throw new TypeError("data must be an object");
  return R(e), t;
}
function Rt(e) {
  const t = {
    "!": "%21",
    "'": "%27",
    "(": "%28",
    ")": "%29",
    "~": "%7E",
    "%20": "+",
    "%00": "\0"
  };
  return encodeURIComponent(e).replace(/[!'()~]|%20|%00/g, function(n) {
    return t[n];
  });
}
function He(e, t) {
  this._pairs = [], e && we(e, this, t);
}
const $t = He.prototype;
$t.append = function(t, r) {
  this._pairs.push([t, r]);
};
$t.toString = function(t) {
  const r = t ? function(n) {
    return t.call(this, n, Rt);
  } : Rt;
  return this._pairs.map(function(o) {
    return r(o[0]) + "=" + r(o[1]);
  }, "").join("&");
};
function wn(e) {
  return encodeURIComponent(e).replace(/%3A/gi, ":").replace(/%24/g, "$").replace(/%2C/gi, ",").replace(/%20/g, "+").replace(/%5B/gi, "[").replace(/%5D/gi, "]");
}
function qt(e, t, r) {
  if (!t)
    return e;
  const n = r && r.encode || wn, o = r && r.serialize;
  let i;
  if (o ? i = o(t, r) : i = l.isURLSearchParams(t) ? t.toString() : new He(t, r).toString(n), i) {
    const a = e.indexOf("#");
    a !== -1 && (e = e.slice(0, a)), e += (e.indexOf("?") === -1 ? "?" : "&") + i;
  }
  return e;
}
class Rn {
  constructor() {
    this.handlers = [];
  }
  /**
   * Add a new interceptor to the stack
   *
   * @param {Function} fulfilled The function to handle `then` for a `Promise`
   * @param {Function} rejected The function to handle `reject` for a `Promise`
   *
   * @return {Number} An ID used to remove interceptor later
   */
  use(t, r, n) {
    return this.handlers.push({
      fulfilled: t,
      rejected: r,
      synchronous: n ? n.synchronous : !1,
      runWhen: n ? n.runWhen : null
    }), this.handlers.length - 1;
  }
  /**
   * Remove an interceptor from the stack
   *
   * @param {Number} id The ID that was returned by `use`
   *
   * @returns {Boolean} `true` if the interceptor was removed, `false` otherwise
   */
  eject(t) {
    this.handlers[t] && (this.handlers[t] = null);
  }
  /**
   * Clear all interceptors from the stack
   *
   * @returns {void}
   */
  clear() {
    this.handlers && (this.handlers = []);
  }
  /**
   * Iterate over all the registered interceptors
   *
   * This method is particularly useful for skipping over any
   * interceptors that may have become `null` calling `eject`.
   *
   * @param {Function} fn The function to call for each interceptor
   *
   * @returns {void}
   */
  forEach(t) {
    l.forEach(this.handlers, function(n) {
      n !== null && t(n);
    });
  }
}
const _t = Rn, Wt = {
  silentJSONParsing: !0,
  forcedJSONParsing: !0,
  clarifyTimeoutError: !1
}, _n = typeof URLSearchParams < "u" ? URLSearchParams : He, On = typeof FormData < "u" ? FormData : null, Sn = typeof Blob < "u" ? Blob : null, xn = (() => {
  let e;
  return typeof navigator < "u" && ((e = navigator.product) === "ReactNative" || e === "NativeScript" || e === "NS") ? !1 : typeof window < "u" && typeof document < "u";
})(), Tn = (() => typeof WorkerGlobalScope < "u" && // eslint-disable-next-line no-undef
self instanceof WorkerGlobalScope && typeof self.importScripts == "function")(), U = {
  isBrowser: !0,
  classes: {
    URLSearchParams: _n,
    FormData: On,
    Blob: Sn
  },
  isStandardBrowserEnv: xn,
  isStandardBrowserWebWorkerEnv: Tn,
  protocols: ["http", "https", "file", "blob", "url", "data"]
};
function Cn(e, t) {
  return we(e, new U.classes.URLSearchParams(), Object.assign({
    visitor: function(r, n, o, i) {
      return U.isNode && l.isBuffer(r) ? (this.append(n, r.toString("base64")), !1) : i.defaultVisitor.apply(this, arguments);
    }
  }, t));
}
function Pn(e) {
  return l.matchAll(/\w+|\[(\w*)]/g, e).map((t) => t[0] === "[]" ? "" : t[1] || t[0]);
}
function An(e) {
  const t = {}, r = Object.keys(e);
  let n;
  const o = r.length;
  let i;
  for (n = 0; n < o; n++)
    i = r[n], t[i] = e[i];
  return t;
}
function Vt(e) {
  function t(r, n, o, i) {
    let a = r[i++];
    const c = Number.isFinite(+a), p = i >= r.length;
    return a = !a && l.isArray(o) ? o.length : a, p ? (l.hasOwnProp(o, a) ? o[a] = [o[a], n] : o[a] = n, !c) : ((!o[a] || !l.isObject(o[a])) && (o[a] = []), t(r, n, o[a], i) && l.isArray(o[a]) && (o[a] = An(o[a])), !c);
  }
  if (l.isFormData(e) && l.isFunction(e.entries)) {
    const r = {};
    return l.forEachEntry(e, (n, o) => {
      t(Pn(n), o, r, 0);
    }), r;
  }
  return null;
}
const jn = {
  "Content-Type": void 0
};
function Nn(e, t, r) {
  if (l.isString(e))
    try {
      return (t || JSON.parse)(e), l.trim(e);
    } catch (n) {
      if (n.name !== "SyntaxError")
        throw n;
    }
  return (r || JSON.stringify)(e);
}
const Re = {
  transitional: Wt,
  adapter: ["xhr", "http"],
  transformRequest: [function(t, r) {
    const n = r.getContentType() || "", o = n.indexOf("application/json") > -1, i = l.isObject(t);
    if (i && l.isHTMLForm(t) && (t = new FormData(t)), l.isFormData(t))
      return o && o ? JSON.stringify(Vt(t)) : t;
    if (l.isArrayBuffer(t) || l.isBuffer(t) || l.isStream(t) || l.isFile(t) || l.isBlob(t))
      return t;
    if (l.isArrayBufferView(t))
      return t.buffer;
    if (l.isURLSearchParams(t))
      return r.setContentType("application/x-www-form-urlencoded;charset=utf-8", !1), t.toString();
    let c;
    if (i) {
      if (n.indexOf("application/x-www-form-urlencoded") > -1)
        return Cn(t, this.formSerializer).toString();
      if ((c = l.isFileList(t)) || n.indexOf("multipart/form-data") > -1) {
        const p = this.env && this.env.FormData;
        return we(
          c ? { "files[]": t } : t,
          p && new p(),
          this.formSerializer
        );
      }
    }
    return i || o ? (r.setContentType("application/json", !1), Nn(t)) : t;
  }],
  transformResponse: [function(t) {
    const r = this.transitional || Re.transitional, n = r && r.forcedJSONParsing, o = this.responseType === "json";
    if (t && l.isString(t) && (n && !this.responseType || o)) {
      const a = !(r && r.silentJSONParsing) && o;
      try {
        return JSON.parse(t);
      } catch (c) {
        if (a)
          throw c.name === "SyntaxError" ? _.from(c, _.ERR_BAD_RESPONSE, this, null, this.response) : c;
      }
    }
    return t;
  }],
  /**
   * A timeout in milliseconds to abort a request. If set to 0 (default) a
   * timeout is not created.
   */
  timeout: 0,
  xsrfCookieName: "XSRF-TOKEN",
  xsrfHeaderName: "X-XSRF-TOKEN",
  maxContentLength: -1,
  maxBodyLength: -1,
  env: {
    FormData: U.classes.FormData,
    Blob: U.classes.Blob
  },
  validateStatus: function(t) {
    return t >= 200 && t < 300;
  },
  headers: {
    common: {
      Accept: "application/json, text/plain, */*"
    }
  }
};
l.forEach(["delete", "get", "head"], function(t) {
  Re.headers[t] = {};
});
l.forEach(["post", "put", "patch"], function(t) {
  Re.headers[t] = l.merge(jn);
});
const $e = Re, Fn = l.toObjectSet([
  "age",
  "authorization",
  "content-length",
  "content-type",
  "etag",
  "expires",
  "from",
  "host",
  "if-modified-since",
  "if-unmodified-since",
  "last-modified",
  "location",
  "max-forwards",
  "proxy-authorization",
  "referer",
  "retry-after",
  "user-agent"
]), Dn = (e) => {
  const t = {};
  let r, n, o;
  return e && e.split(`
`).forEach(function(a) {
    o = a.indexOf(":"), r = a.substring(0, o).trim().toLowerCase(), n = a.substring(o + 1).trim(), !(!r || t[r] && Fn[r]) && (r === "set-cookie" ? t[r] ? t[r].push(n) : t[r] = [n] : t[r] = t[r] ? t[r] + ", " + n : n);
  }), t;
}, Ot = Symbol("internals");
function ne(e) {
  return e && String(e).trim().toLowerCase();
}
function he(e) {
  return e === !1 || e == null ? e : l.isArray(e) ? e.map(he) : String(e);
}
function kn(e) {
  const t = /* @__PURE__ */ Object.create(null), r = /([^\s,;=]+)\s*(?:=\s*([^,;]+))?/g;
  let n;
  for (; n = r.exec(e); )
    t[n[1]] = n[2];
  return t;
}
const Bn = (e) => /^[-_a-zA-Z0-9^`|~,!#$%&'*+.]+$/.test(e.trim());
function Fe(e, t, r, n, o) {
  if (l.isFunction(n))
    return n.call(this, t, r);
  if (o && (t = r), !!l.isString(t)) {
    if (l.isString(n))
      return t.indexOf(n) !== -1;
    if (l.isRegExp(n))
      return n.test(t);
  }
}
function Ln(e) {
  return e.trim().toLowerCase().replace(/([a-z\d])(\w*)/g, (t, r, n) => r.toUpperCase() + n);
}
function Un(e, t) {
  const r = l.toCamelCase(" " + t);
  ["get", "set", "has"].forEach((n) => {
    Object.defineProperty(e, n + r, {
      value: function(o, i, a) {
        return this[n].call(this, t, o, i, a);
      },
      configurable: !0
    });
  });
}
class _e {
  constructor(t) {
    t && this.set(t);
  }
  set(t, r, n) {
    const o = this;
    function i(c, p, u) {
      const d = ne(p);
      if (!d)
        throw new Error("header name must be a non-empty string");
      const y = l.findKey(o, d);
      (!y || o[y] === void 0 || u === !0 || u === void 0 && o[y] !== !1) && (o[y || p] = he(c));
    }
    const a = (c, p) => l.forEach(c, (u, d) => i(u, d, p));
    return l.isPlainObject(t) || t instanceof this.constructor ? a(t, r) : l.isString(t) && (t = t.trim()) && !Bn(t) ? a(Dn(t), r) : t != null && i(r, t, n), this;
  }
  get(t, r) {
    if (t = ne(t), t) {
      const n = l.findKey(this, t);
      if (n) {
        const o = this[n];
        if (!r)
          return o;
        if (r === !0)
          return kn(o);
        if (l.isFunction(r))
          return r.call(this, o, n);
        if (l.isRegExp(r))
          return r.exec(o);
        throw new TypeError("parser must be boolean|regexp|function");
      }
    }
  }
  has(t, r) {
    if (t = ne(t), t) {
      const n = l.findKey(this, t);
      return !!(n && this[n] !== void 0 && (!r || Fe(this, this[n], n, r)));
    }
    return !1;
  }
  delete(t, r) {
    const n = this;
    let o = !1;
    function i(a) {
      if (a = ne(a), a) {
        const c = l.findKey(n, a);
        c && (!r || Fe(n, n[c], c, r)) && (delete n[c], o = !0);
      }
    }
    return l.isArray(t) ? t.forEach(i) : i(t), o;
  }
  clear(t) {
    const r = Object.keys(this);
    let n = r.length, o = !1;
    for (; n--; ) {
      const i = r[n];
      (!t || Fe(this, this[i], i, t, !0)) && (delete this[i], o = !0);
    }
    return o;
  }
  normalize(t) {
    const r = this, n = {};
    return l.forEach(this, (o, i) => {
      const a = l.findKey(n, i);
      if (a) {
        r[a] = he(o), delete r[i];
        return;
      }
      const c = t ? Ln(i) : String(i).trim();
      c !== i && delete r[i], r[c] = he(o), n[c] = !0;
    }), this;
  }
  concat(...t) {
    return this.constructor.concat(this, ...t);
  }
  toJSON(t) {
    const r = /* @__PURE__ */ Object.create(null);
    return l.forEach(this, (n, o) => {
      n != null && n !== !1 && (r[o] = t && l.isArray(n) ? n.join(", ") : n);
    }), r;
  }
  [Symbol.iterator]() {
    return Object.entries(this.toJSON())[Symbol.iterator]();
  }
  toString() {
    return Object.entries(this.toJSON()).map(([t, r]) => t + ": " + r).join(`
`);
  }
  get [Symbol.toStringTag]() {
    return "AxiosHeaders";
  }
  static from(t) {
    return t instanceof this ? t : new this(t);
  }
  static concat(t, ...r) {
    const n = new this(t);
    return r.forEach((o) => n.set(o)), n;
  }
  static accessor(t) {
    const n = (this[Ot] = this[Ot] = {
      accessors: {}
    }).accessors, o = this.prototype;
    function i(a) {
      const c = ne(a);
      n[c] || (Un(o, a), n[c] = !0);
    }
    return l.isArray(t) ? t.forEach(i) : i(t), this;
  }
}
_e.accessor(["Content-Type", "Content-Length", "Accept", "Accept-Encoding", "User-Agent", "Authorization"]);
l.freezeMethods(_e.prototype);
l.freezeMethods(_e);
const $ = _e;
function De(e, t) {
  const r = this || $e, n = t || r, o = $.from(n.headers);
  let i = n.data;
  return l.forEach(e, function(c) {
    i = c.call(r, i, o.normalize(), t ? t.status : void 0);
  }), o.normalize(), i;
}
function Jt(e) {
  return !!(e && e.__CANCEL__);
}
function ae(e, t, r) {
  _.call(this, e ?? "canceled", _.ERR_CANCELED, t, r), this.name = "CanceledError";
}
l.inherits(ae, _, {
  __CANCEL__: !0
});
function In(e, t, r) {
  const n = r.config.validateStatus;
  !r.status || !n || n(r.status) ? e(r) : t(new _(
    "Request failed with status code " + r.status,
    [_.ERR_BAD_REQUEST, _.ERR_BAD_RESPONSE][Math.floor(r.status / 100) - 4],
    r.config,
    r.request,
    r
  ));
}
const Mn = U.isStandardBrowserEnv ? (
  // Standard browser envs support document.cookie
  function() {
    return {
      write: function(r, n, o, i, a, c) {
        const p = [];
        p.push(r + "=" + encodeURIComponent(n)), l.isNumber(o) && p.push("expires=" + new Date(o).toGMTString()), l.isString(i) && p.push("path=" + i), l.isString(a) && p.push("domain=" + a), c === !0 && p.push("secure"), document.cookie = p.join("; ");
      },
      read: function(r) {
        const n = document.cookie.match(new RegExp("(^|;\\s*)(" + r + ")=([^;]*)"));
        return n ? decodeURIComponent(n[3]) : null;
      },
      remove: function(r) {
        this.write(r, "", Date.now() - 864e5);
      }
    };
  }()
) : (
  // Non standard browser env (web workers, react-native) lack needed support.
  function() {
    return {
      write: function() {
      },
      read: function() {
        return null;
      },
      remove: function() {
      }
    };
  }()
);
function zn(e) {
  return /^([a-z][a-z\d+\-.]*:)?\/\//i.test(e);
}
function Hn(e, t) {
  return t ? e.replace(/\/+$/, "") + "/" + t.replace(/^\/+/, "") : e;
}
function Yt(e, t) {
  return e && !zn(t) ? Hn(e, t) : t;
}
const $n = U.isStandardBrowserEnv ? (
  // Standard browser envs have full support of the APIs needed to test
  // whether the request URL is of the same origin as current location.
  function() {
    const t = /(msie|trident)/i.test(navigator.userAgent), r = document.createElement("a");
    let n;
    function o(i) {
      let a = i;
      return t && (r.setAttribute("href", a), a = r.href), r.setAttribute("href", a), {
        href: r.href,
        protocol: r.protocol ? r.protocol.replace(/:$/, "") : "",
        host: r.host,
        search: r.search ? r.search.replace(/^\?/, "") : "",
        hash: r.hash ? r.hash.replace(/^#/, "") : "",
        hostname: r.hostname,
        port: r.port,
        pathname: r.pathname.charAt(0) === "/" ? r.pathname : "/" + r.pathname
      };
    }
    return n = o(window.location.href), function(a) {
      const c = l.isString(a) ? o(a) : a;
      return c.protocol === n.protocol && c.host === n.host;
    };
  }()
) : (
  // Non standard browser envs (web workers, react-native) lack needed support.
  function() {
    return function() {
      return !0;
    };
  }()
);
function qn(e) {
  const t = /^([-+\w]{1,25})(:?\/\/|:)/.exec(e);
  return t && t[1] || "";
}
function Wn(e, t) {
  e = e || 10;
  const r = new Array(e), n = new Array(e);
  let o = 0, i = 0, a;
  return t = t !== void 0 ? t : 1e3, function(p) {
    const u = Date.now(), d = n[i];
    a || (a = u), r[o] = p, n[o] = u;
    let y = i, w = 0;
    for (; y !== o; )
      w += r[y++], y = y % e;
    if (o = (o + 1) % e, o === i && (i = (i + 1) % e), u - a < t)
      return;
    const R = d && u - d;
    return R ? Math.round(w * 1e3 / R) : void 0;
  };
}
function St(e, t) {
  let r = 0;
  const n = Wn(50, 250);
  return (o) => {
    const i = o.loaded, a = o.lengthComputable ? o.total : void 0, c = i - r, p = n(c), u = i <= a;
    r = i;
    const d = {
      loaded: i,
      total: a,
      progress: a ? i / a : void 0,
      bytes: c,
      rate: p || void 0,
      estimated: p && a && u ? (a - i) / p : void 0,
      event: o
    };
    d[t ? "download" : "upload"] = !0, e(d);
  };
}
const Vn = typeof XMLHttpRequest < "u", Jn = Vn && function(e) {
  return new Promise(function(r, n) {
    let o = e.data;
    const i = $.from(e.headers).normalize(), a = e.responseType;
    let c;
    function p() {
      e.cancelToken && e.cancelToken.unsubscribe(c), e.signal && e.signal.removeEventListener("abort", c);
    }
    l.isFormData(o) && (U.isStandardBrowserEnv || U.isStandardBrowserWebWorkerEnv ? i.setContentType(!1) : i.setContentType("multipart/form-data;", !1));
    let u = new XMLHttpRequest();
    if (e.auth) {
      const R = e.auth.username || "", m = e.auth.password ? unescape(encodeURIComponent(e.auth.password)) : "";
      i.set("Authorization", "Basic " + btoa(R + ":" + m));
    }
    const d = Yt(e.baseURL, e.url);
    u.open(e.method.toUpperCase(), qt(d, e.params, e.paramsSerializer), !0), u.timeout = e.timeout;
    function y() {
      if (!u)
        return;
      const R = $.from(
        "getAllResponseHeaders" in u && u.getAllResponseHeaders()
      ), b = {
        data: !a || a === "text" || a === "json" ? u.responseText : u.response,
        status: u.status,
        statusText: u.statusText,
        headers: R,
        config: e,
        request: u
      };
      In(function(C) {
        r(C), p();
      }, function(C) {
        n(C), p();
      }, b), u = null;
    }
    if ("onloadend" in u ? u.onloadend = y : u.onreadystatechange = function() {
      !u || u.readyState !== 4 || u.status === 0 && !(u.responseURL && u.responseURL.indexOf("file:") === 0) || setTimeout(y);
    }, u.onabort = function() {
      u && (n(new _("Request aborted", _.ECONNABORTED, e, u)), u = null);
    }, u.onerror = function() {
      n(new _("Network Error", _.ERR_NETWORK, e, u)), u = null;
    }, u.ontimeout = function() {
      let m = e.timeout ? "timeout of " + e.timeout + "ms exceeded" : "timeout exceeded";
      const b = e.transitional || Wt;
      e.timeoutErrorMessage && (m = e.timeoutErrorMessage), n(new _(
        m,
        b.clarifyTimeoutError ? _.ETIMEDOUT : _.ECONNABORTED,
        e,
        u
      )), u = null;
    }, U.isStandardBrowserEnv) {
      const R = (e.withCredentials || $n(d)) && e.xsrfCookieName && Mn.read(e.xsrfCookieName);
      R && i.set(e.xsrfHeaderName, R);
    }
    o === void 0 && i.setContentType(null), "setRequestHeader" in u && l.forEach(i.toJSON(), function(m, b) {
      u.setRequestHeader(b, m);
    }), l.isUndefined(e.withCredentials) || (u.withCredentials = !!e.withCredentials), a && a !== "json" && (u.responseType = e.responseType), typeof e.onDownloadProgress == "function" && u.addEventListener("progress", St(e.onDownloadProgress, !0)), typeof e.onUploadProgress == "function" && u.upload && u.upload.addEventListener("progress", St(e.onUploadProgress)), (e.cancelToken || e.signal) && (c = (R) => {
      u && (n(!R || R.type ? new ae(null, e, u) : R), u.abort(), u = null);
    }, e.cancelToken && e.cancelToken.subscribe(c), e.signal && (e.signal.aborted ? c() : e.signal.addEventListener("abort", c)));
    const w = qn(d);
    if (w && U.protocols.indexOf(w) === -1) {
      n(new _("Unsupported protocol " + w + ":", _.ERR_BAD_REQUEST, e));
      return;
    }
    u.send(o || null);
  });
}, me = {
  http: bn,
  xhr: Jn
};
l.forEach(me, (e, t) => {
  if (e) {
    try {
      Object.defineProperty(e, "name", { value: t });
    } catch {
    }
    Object.defineProperty(e, "adapterName", { value: t });
  }
});
const Yn = {
  getAdapter: (e) => {
    e = l.isArray(e) ? e : [e];
    const { length: t } = e;
    let r, n;
    for (let o = 0; o < t && (r = e[o], !(n = l.isString(r) ? me[r.toLowerCase()] : r)); o++)
      ;
    if (!n)
      throw n === !1 ? new _(
        `Adapter ${r} is not supported by the environment`,
        "ERR_NOT_SUPPORT"
      ) : new Error(
        l.hasOwnProp(me, r) ? `Adapter '${r}' is not available in the build` : `Unknown adapter '${r}'`
      );
    if (!l.isFunction(n))
      throw new TypeError("adapter is not a function");
    return n;
  },
  adapters: me
};
function ke(e) {
  if (e.cancelToken && e.cancelToken.throwIfRequested(), e.signal && e.signal.aborted)
    throw new ae(null, e);
}
function xt(e) {
  return ke(e), e.headers = $.from(e.headers), e.data = De.call(
    e,
    e.transformRequest
  ), ["post", "put", "patch"].indexOf(e.method) !== -1 && e.headers.setContentType("application/x-www-form-urlencoded", !1), Yn.getAdapter(e.adapter || $e.adapter)(e).then(function(n) {
    return ke(e), n.data = De.call(
      e,
      e.transformResponse,
      n
    ), n.headers = $.from(n.headers), n;
  }, function(n) {
    return Jt(n) || (ke(e), n && n.response && (n.response.data = De.call(
      e,
      e.transformResponse,
      n.response
    ), n.response.headers = $.from(n.response.headers))), Promise.reject(n);
  });
}
const Tt = (e) => e instanceof $ ? e.toJSON() : e;
function X(e, t) {
  t = t || {};
  const r = {};
  function n(u, d, y) {
    return l.isPlainObject(u) && l.isPlainObject(d) ? l.merge.call({ caseless: y }, u, d) : l.isPlainObject(d) ? l.merge({}, d) : l.isArray(d) ? d.slice() : d;
  }
  function o(u, d, y) {
    if (l.isUndefined(d)) {
      if (!l.isUndefined(u))
        return n(void 0, u, y);
    } else
      return n(u, d, y);
  }
  function i(u, d) {
    if (!l.isUndefined(d))
      return n(void 0, d);
  }
  function a(u, d) {
    if (l.isUndefined(d)) {
      if (!l.isUndefined(u))
        return n(void 0, u);
    } else
      return n(void 0, d);
  }
  function c(u, d, y) {
    if (y in t)
      return n(u, d);
    if (y in e)
      return n(void 0, u);
  }
  const p = {
    url: i,
    method: i,
    data: i,
    baseURL: a,
    transformRequest: a,
    transformResponse: a,
    paramsSerializer: a,
    timeout: a,
    timeoutMessage: a,
    withCredentials: a,
    adapter: a,
    responseType: a,
    xsrfCookieName: a,
    xsrfHeaderName: a,
    onUploadProgress: a,
    onDownloadProgress: a,
    decompress: a,
    maxContentLength: a,
    maxBodyLength: a,
    beforeRedirect: a,
    transport: a,
    httpAgent: a,
    httpsAgent: a,
    cancelToken: a,
    socketPath: a,
    responseEncoding: a,
    validateStatus: c,
    headers: (u, d) => o(Tt(u), Tt(d), !0)
  };
  return l.forEach(Object.keys(Object.assign({}, e, t)), function(d) {
    const y = p[d] || o, w = y(e[d], t[d], d);
    l.isUndefined(w) && y !== c || (r[d] = w);
  }), r;
}
const Kt = "1.4.0", qe = {};
["object", "boolean", "number", "function", "string", "symbol"].forEach((e, t) => {
  qe[e] = function(n) {
    return typeof n === e || "a" + (t < 1 ? "n " : " ") + e;
  };
});
const Ct = {};
qe.transitional = function(t, r, n) {
  function o(i, a) {
    return "[Axios v" + Kt + "] Transitional option '" + i + "'" + a + (n ? ". " + n : "");
  }
  return (i, a, c) => {
    if (t === !1)
      throw new _(
        o(a, " has been removed" + (r ? " in " + r : "")),
        _.ERR_DEPRECATED
      );
    return r && !Ct[a] && (Ct[a] = !0, console.warn(
      o(
        a,
        " has been deprecated since v" + r + " and will be removed in the near future"
      )
    )), t ? t(i, a, c) : !0;
  };
};
function Kn(e, t, r) {
  if (typeof e != "object")
    throw new _("options must be an object", _.ERR_BAD_OPTION_VALUE);
  const n = Object.keys(e);
  let o = n.length;
  for (; o-- > 0; ) {
    const i = n[o], a = t[i];
    if (a) {
      const c = e[i], p = c === void 0 || a(c, i, e);
      if (p !== !0)
        throw new _("option " + i + " must be " + p, _.ERR_BAD_OPTION_VALUE);
      continue;
    }
    if (r !== !0)
      throw new _("Unknown option " + i, _.ERR_BAD_OPTION);
  }
}
const Ie = {
  assertOptions: Kn,
  validators: qe
}, q = Ie.validators;
class Ee {
  constructor(t) {
    this.defaults = t, this.interceptors = {
      request: new _t(),
      response: new _t()
    };
  }
  /**
   * Dispatch a request
   *
   * @param {String|Object} configOrUrl The config specific for this request (merged with this.defaults)
   * @param {?Object} config
   *
   * @returns {Promise} The Promise to be fulfilled
   */
  request(t, r) {
    typeof t == "string" ? (r = r || {}, r.url = t) : r = t || {}, r = X(this.defaults, r);
    const { transitional: n, paramsSerializer: o, headers: i } = r;
    n !== void 0 && Ie.assertOptions(n, {
      silentJSONParsing: q.transitional(q.boolean),
      forcedJSONParsing: q.transitional(q.boolean),
      clarifyTimeoutError: q.transitional(q.boolean)
    }, !1), o != null && (l.isFunction(o) ? r.paramsSerializer = {
      serialize: o
    } : Ie.assertOptions(o, {
      encode: q.function,
      serialize: q.function
    }, !0)), r.method = (r.method || this.defaults.method || "get").toLowerCase();
    let a;
    a = i && l.merge(
      i.common,
      i[r.method]
    ), a && l.forEach(
      ["delete", "get", "head", "post", "put", "patch", "common"],
      (m) => {
        delete i[m];
      }
    ), r.headers = $.concat(a, i);
    const c = [];
    let p = !0;
    this.interceptors.request.forEach(function(b) {
      typeof b.runWhen == "function" && b.runWhen(r) === !1 || (p = p && b.synchronous, c.unshift(b.fulfilled, b.rejected));
    });
    const u = [];
    this.interceptors.response.forEach(function(b) {
      u.push(b.fulfilled, b.rejected);
    });
    let d, y = 0, w;
    if (!p) {
      const m = [xt.bind(this), void 0];
      for (m.unshift.apply(m, c), m.push.apply(m, u), w = m.length, d = Promise.resolve(r); y < w; )
        d = d.then(m[y++], m[y++]);
      return d;
    }
    w = c.length;
    let R = r;
    for (y = 0; y < w; ) {
      const m = c[y++], b = c[y++];
      try {
        R = m(R);
      } catch (D) {
        b.call(this, D);
        break;
      }
    }
    try {
      d = xt.call(this, R);
    } catch (m) {
      return Promise.reject(m);
    }
    for (y = 0, w = u.length; y < w; )
      d = d.then(u[y++], u[y++]);
    return d;
  }
  getUri(t) {
    t = X(this.defaults, t);
    const r = Yt(t.baseURL, t.url);
    return qt(r, t.params, t.paramsSerializer);
  }
}
l.forEach(["delete", "get", "head", "options"], function(t) {
  Ee.prototype[t] = function(r, n) {
    return this.request(X(n || {}, {
      method: t,
      url: r,
      data: (n || {}).data
    }));
  };
});
l.forEach(["post", "put", "patch"], function(t) {
  function r(n) {
    return function(i, a, c) {
      return this.request(X(c || {}, {
        method: t,
        headers: n ? {
          "Content-Type": "multipart/form-data"
        } : {},
        url: i,
        data: a
      }));
    };
  }
  Ee.prototype[t] = r(), Ee.prototype[t + "Form"] = r(!0);
});
const ye = Ee;
class We {
  constructor(t) {
    if (typeof t != "function")
      throw new TypeError("executor must be a function.");
    let r;
    this.promise = new Promise(function(i) {
      r = i;
    });
    const n = this;
    this.promise.then((o) => {
      if (!n._listeners)
        return;
      let i = n._listeners.length;
      for (; i-- > 0; )
        n._listeners[i](o);
      n._listeners = null;
    }), this.promise.then = (o) => {
      let i;
      const a = new Promise((c) => {
        n.subscribe(c), i = c;
      }).then(o);
      return a.cancel = function() {
        n.unsubscribe(i);
      }, a;
    }, t(function(i, a, c) {
      n.reason || (n.reason = new ae(i, a, c), r(n.reason));
    });
  }
  /**
   * Throws a `CanceledError` if cancellation has been requested.
   */
  throwIfRequested() {
    if (this.reason)
      throw this.reason;
  }
  /**
   * Subscribe to the cancel signal
   */
  subscribe(t) {
    if (this.reason) {
      t(this.reason);
      return;
    }
    this._listeners ? this._listeners.push(t) : this._listeners = [t];
  }
  /**
   * Unsubscribe from the cancel signal
   */
  unsubscribe(t) {
    if (!this._listeners)
      return;
    const r = this._listeners.indexOf(t);
    r !== -1 && this._listeners.splice(r, 1);
  }
  /**
   * Returns an object that contains a new `CancelToken` and a function that, when called,
   * cancels the `CancelToken`.
   */
  static source() {
    let t;
    return {
      token: new We(function(o) {
        t = o;
      }),
      cancel: t
    };
  }
}
const Gn = We;
function Xn(e) {
  return function(r) {
    return e.apply(null, r);
  };
}
function Zn(e) {
  return l.isObject(e) && e.isAxiosError === !0;
}
const Me = {
  Continue: 100,
  SwitchingProtocols: 101,
  Processing: 102,
  EarlyHints: 103,
  Ok: 200,
  Created: 201,
  Accepted: 202,
  NonAuthoritativeInformation: 203,
  NoContent: 204,
  ResetContent: 205,
  PartialContent: 206,
  MultiStatus: 207,
  AlreadyReported: 208,
  ImUsed: 226,
  MultipleChoices: 300,
  MovedPermanently: 301,
  Found: 302,
  SeeOther: 303,
  NotModified: 304,
  UseProxy: 305,
  Unused: 306,
  TemporaryRedirect: 307,
  PermanentRedirect: 308,
  BadRequest: 400,
  Unauthorized: 401,
  PaymentRequired: 402,
  Forbidden: 403,
  NotFound: 404,
  MethodNotAllowed: 405,
  NotAcceptable: 406,
  ProxyAuthenticationRequired: 407,
  RequestTimeout: 408,
  Conflict: 409,
  Gone: 410,
  LengthRequired: 411,
  PreconditionFailed: 412,
  PayloadTooLarge: 413,
  UriTooLong: 414,
  UnsupportedMediaType: 415,
  RangeNotSatisfiable: 416,
  ExpectationFailed: 417,
  ImATeapot: 418,
  MisdirectedRequest: 421,
  UnprocessableEntity: 422,
  Locked: 423,
  FailedDependency: 424,
  TooEarly: 425,
  UpgradeRequired: 426,
  PreconditionRequired: 428,
  TooManyRequests: 429,
  RequestHeaderFieldsTooLarge: 431,
  UnavailableForLegalReasons: 451,
  InternalServerError: 500,
  NotImplemented: 501,
  BadGateway: 502,
  ServiceUnavailable: 503,
  GatewayTimeout: 504,
  HttpVersionNotSupported: 505,
  VariantAlsoNegotiates: 506,
  InsufficientStorage: 507,
  LoopDetected: 508,
  NotExtended: 510,
  NetworkAuthenticationRequired: 511
};
Object.entries(Me).forEach(([e, t]) => {
  Me[t] = e;
});
const Qn = Me;
function Gt(e) {
  const t = new ye(e), r = Nt(ye.prototype.request, t);
  return l.extend(r, ye.prototype, t, { allOwnKeys: !0 }), l.extend(r, t, null, { allOwnKeys: !0 }), r.create = function(o) {
    return Gt(X(e, o));
  }, r;
}
const j = Gt($e);
j.Axios = ye;
j.CanceledError = ae;
j.CancelToken = Gn;
j.isCancel = Jt;
j.VERSION = Kt;
j.toFormData = we;
j.AxiosError = _;
j.Cancel = j.CanceledError;
j.all = function(t) {
  return Promise.all(t);
};
j.spread = Xn;
j.isAxiosError = Zn;
j.mergeConfig = X;
j.AxiosHeaders = $;
j.formToJSON = (e) => Vt(l.isHTMLForm(e) ? new FormData(e) : e);
j.HttpStatusCode = Qn;
j.default = j;
const es = j;
function ts(e) {
  return oe({ tag: "svg", attr: { viewBox: "0 0 24 24", fill: "none" }, child: [{ tag: "path", attr: { opacity: "0.2", fillRule: "evenodd", clipRule: "evenodd", d: "M12 19C15.866 19 19 15.866 19 12C19 8.13401 15.866 5 12 5C8.13401 5 5 8.13401 5 12C5 15.866 8.13401 19 12 19ZM12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z", fill: "currentColor" } }, { tag: "path", attr: { d: "M2 12C2 6.47715 6.47715 2 12 2V5C8.13401 5 5 8.13401 5 12H2Z", fill: "currentColor" } }] })(e);
}
function rs(e) {
  return oe({ tag: "svg", attr: { fill: "currentColor", viewBox: "0 0 16 16" }, child: [{ tag: "path", attr: { fillRule: "evenodd", d: "M3.112 5.112a3.125 3.125 0 0 0-.17.613C1.266 6.095 0 7.555 0 9.318 0 11.366 1.708 13 3.781 13H11L3.112 5.112zm11.372 7.372L4.937 2.937A5.512 5.512 0 0 1 8 2c2.69 0 4.923 2 5.166 4.579C14.758 6.804 16 8.137 16 9.773a3.2 3.2 0 0 1-1.516 2.711zm-.838 1.87-12-12 .708-.708 12 12-.707.707z" } }] })(e);
}
function ns(e) {
  return oe({ tag: "svg", attr: { fill: "currentColor", viewBox: "0 0 16 16" }, child: [{ tag: "path", attr: { d: "M8 2a5.53 5.53 0 0 0-3.594 1.342c-.766.66-1.321 1.52-1.464 2.383C1.266 6.095 0 7.555 0 9.318 0 11.366 1.708 13 3.781 13h8.906C14.502 13 16 11.57 16 9.773c0-1.636-1.242-2.969-2.834-3.194C12.923 3.999 10.69 2 8 2zm2.354 4.854-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 1 1 .708-.708L7 8.793l2.646-2.647a.5.5 0 0 1 .708.708z" } }] })(e);
}
const os = ({
  children: e,
  apiKey: t,
  hideAttribution: r,
  onUploadComplete: n,
  accentColor: o = "#f35815"
}) => {
  const [i, a] = G(!1), c = Tr(null), [p, u] = G(null), [d, y] = G(!1), [w, R] = G(!1), [m, b] = G(!1), D = async () => {
    const N = new FormData();
    N.append("file", p);
    try {
      y(!0);
      const { data: J } = await es.post(
        "https://api.uploadfly.cloud/upload",
        N,
        {
          headers: {
            Authorization: `Bearer ${t}`
          }
        }
      );
      R(!0), n && n(J);
    } catch {
      b(!0);
    } finally {
      y(!1);
    }
  }, C = () => {
    d || (a(!1), u(null), b(!1), R(!1));
  }, [T, L] = G(!1);
  return /* @__PURE__ */ O.jsxs(
    "div",
    {
      className: "uf_widget_container",
      style: { "--accent": o },
      onDragOver: (N) => {
        N.preventDefault(), L(!0);
      },
      onDragLeave: (N) => {
        N.preventDefault(), L(!1);
      },
      onDrop: (N) => {
        N.preventDefault(), L(!1);
        const J = N.dataTransfer.files[0];
        u(J);
      },
      children: [
        /* @__PURE__ */ O.jsx("div", { onClick: () => a(!0), children: e }),
        i && /* @__PURE__ */ O.jsx(
          "div",
          {
            className: "uf_widget_modal",
            style: {
              pointerEvents: i ? "auto" : "none"
            },
            children: /* @__PURE__ */ O.jsx("div", { className: "uf_modal_backdrop rounded-md", onClick: C, children: /* @__PURE__ */ O.jsxs(
              "div",
              {
                className: `uf_modal_container ${T ? "drag-over" : ""}`,
                onClick: (N) => N.stopPropagation(),
                children: [
                  /* @__PURE__ */ O.jsx("button", { onClick: C, className: "uf_close_button", children: /* @__PURE__ */ O.jsx(Dr, {}) }),
                  /* @__PURE__ */ O.jsxs(
                    "div",
                    {
                      style: {
                        width: "100%"
                      },
                      children: [
                        /* @__PURE__ */ O.jsx(
                          "input",
                          {
                            type: "file",
                            ref: c,
                            hidden: !0,
                            multiple: !1,
                            onChange: (N) => u(N.target.files[0])
                          }
                        ),
                        !p && !m && !w && /* @__PURE__ */ O.jsxs("div", { className: "uf_choose_a_file", children: [
                          /* @__PURE__ */ O.jsx(
                            "button",
                            {
                              className: "uf_choose_a_file_btn",
                              onClick: () => c.current.click(),
                              children: "Choose a file"
                            }
                          ),
                          /* @__PURE__ */ O.jsx("p", { className: "uf_choose_a_file_text", children: "...or drop a file" })
                        ] }),
                        p && !m && !w && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_selected", children: [
                          !d && /* @__PURE__ */ O.jsx(
                            "button",
                            {
                              className: "uf_change_file",
                              onClick: () => c.current.click(),
                              children: "Change file"
                            }
                          ),
                          /* @__PURE__ */ O.jsxs("div", { className: "uf_file_info", children: [
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Name: " }),
                              p.name
                            ] }),
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Type: " }),
                              p.type
                            ] }),
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Size: " }),
                              Br(p.size).human("si")
                            ] })
                          ] }),
                          /* @__PURE__ */ O.jsxs(
                            "button",
                            {
                              className: "uf_upload_button",
                              disabled: d,
                              onClick: D,
                              children: [
                                d ? /* @__PURE__ */ O.jsx(ts, { className: "uf_animate_spin" }) : /* @__PURE__ */ O.jsx(Fr, {}),
                                "Upload",
                                d && "ing..."
                              ]
                            }
                          )
                        ] }),
                        m && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_upload_response", children: [
                          /* @__PURE__ */ O.jsx(
                            rs,
                            {
                              style: {
                                color: "rgb(220, 38, 38)",
                                fontSize: "4.5rem",
                                lineHeight: 1
                              }
                            }
                          ),
                          /* @__PURE__ */ O.jsx("h2", { className: "", children: "File upload failed" })
                        ] }),
                        w && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_upload_response", children: [
                          /* @__PURE__ */ O.jsx(
                            ns,
                            {
                              style: {
                                color: "rgb(22, 163, 74)",
                                fontSize: "4.5rem",
                                lineHeight: 1
                              }
                            }
                          ),
                          /* @__PURE__ */ O.jsx("h2", { children: "File upload successful" })
                        ] })
                      ]
                    }
                  ),
                  !r && /* @__PURE__ */ O.jsxs("p", { className: "uf_attribution", children: [
                    "Powered with  by",
                    " ",
                    /* @__PURE__ */ O.jsx(
                      "a",
                      {
                        href: "https://uploadfly.cloud?utm_source=react_widget",
                        target: "_blank",
                        children: "Uploadfly"
                      }
                    )
                  ] })
                ]
              }
            ) })
          }
        )
      ]
    }
  );
};
export {
  os as UfWidget
};
